### PR TITLE
docs: update extproc stats

### PR DESCRIPTION
**Pull request by tsukiy0**
Signed-off-by: Tony Tang <tsukiy0@users.noreply.github.com>

Commit Message: update extproc stats
Additional Description: added missing `message_timeouts` and `rejected_header_mutations` stats to ext_proc docs
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: docs/root/configuration/http/http_filters/ext_proc_filter.rst [hunk 1]
```diff
@@ -48,3 +48,5 @@ The following statistics are supported:
   streams_closed, Counter, The number of streams successfully closed on either end
   streams_failed, Counter, The number of times a stream produced a gRPC error
   failure_mode_allowed, Counter, The number of times an error was ignored due to configuration
+  message_timeouts, Counter, The number of times a message fails to receive a response within the configured timeout
```
Comment (daixiang0) [inline docs/root/configuration/http/http_filters/ext_proc_filter.rst:51]: small nit, fails -> failed.

File: docs/root/configuration/http/http_filters/ext_proc_filter.rst [hunk 2]
```diff
@@ -48,3 +48,5 @@ The following statistics are supported:
...
+  rejected_header_mutations, Counter, The number of attempts to change diallowed headers
```
Comment (daixiang0) [inline docs/root/configuration/http/http_filters/ext_proc_filter.rst:52]: Maybe highlight reject is better: `The number of rejected header mutations`.

**Comment by mattklein123** [comment]
cc @gbrail 

This suggested changes make sense to me. Thank you!

/wait

**Review by mattklein123 — APPROVED** [review]
Thanks!

**Pull request merged**